pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -317,19 +242,57 @@ public static String recoverSpecialCharacterInPropertyValue(String value) {
     return value.replace("\0\0", ",");
   }
 
+  /**
```

Comment:
Moved the function [getStringListFromSegmentProperties](https://github.com/apache/pinot/blob/master/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/LoaderUtils.java#L70C30-L70C64) to here, since commons-configuration2 saves list-based data individually, for exampel
```bash
segment.dimension.column.names = bytesDimSV1
segment.dimension.column.names = generationNumber
segment.dimension.column.names = intDimMV1
segment.dimension.column.names = intDimMV2
segment.dimension.column.names = longDimSV1
segment.dimension.column.names = longDimSV2
segment.dimension.column.names = mapDim1__KEYS
segment.dimension.column.names = mapDim1__VALUES
segment.dimension.column.names = mapDim2json
segment.dimension.column.names = stringDimMV1
segment.dimension.column.names = stringDimMV2
segment.dimension.column.names = stringDimSV1
segment.dimension.column.names = stringDimSV2
segment.dimension.column.names = textDim1
segment.metric.column.names = doubleMetric1
segment.metric.column.names = floatMetric1
segment.metric.column.names = intMetric1
segment.metric.column.names = longMetric1
```
to make this migration compatible, enhanced the function to split the values in-case if it has comma `,`

Reply:
Do we need this? After setting the list delimiter, will `getList()` automatically treat `,` as list delimiter?
If not, this is high risk change because other developer might not know they cannot use `getList()` to retrieve the list values

Reply:
They can use `getList()` without any issue with this change. It's just setting COMMA `,` which results in weird escaping issues and results in failure for values with leading or trailing spaces or whitespace. 
After certain iterations, I got this working with backward compatibility.

Reply:
Sorry I still don't fully follow. Is there behavior change when calling `getList()` on comma separated list? Why do we need to explicitly split the value here?

Reply:
I'll try to explain it to you.

In commons-configuration1, we were setting Comma `,` as ListDelimiter. The metadata files written with this property result in writing the list values in one line separated by a comma. For string values having the comma, we are[ replacing it with](https://github.com/apache/pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java#L300C31-L300C37) `\0\0`

The same setup with commons-configuration2 is working differently. Commons-configuration2 internally changed the implementation of escaping/unescaping the values. I have witnessed the issue of handling string values with leading or trailing whitespace. 

I got the upgrade working with backward compatibility by not setting Comma `,` as a list delimiter. Since there is no delimiter set, commons-configuration2 saves the list property values individually in a separate line, for example like this
```bash
segment.dimension.column.names = bytesDimSV1
segment.dimension.column.names = generationNumber
segment.dimension.column.names = intDimMV1
segment.dimension.column.names = intDimMV2
segment.dimension.column.names = longDimSV1
```
And, to make it backward compatible,  we are splitting the list values based on comma `,`.  

for escaping and unescaping we are now using `StringEscapeUtils` functionality.  Please, let me know if this looks correct to you. Thanks

Reply:
Instead of using [DefaultListDelimiterHandler](https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/convert/DefaultListDelimiterHandler.html), [LegacyListDelimiterHandler](https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/convert/LegacyListDelimiterHandler.html) follows the commons-configuration1 implementation. Reverting the changes related to spliting the string values.